crates/router/src/core/blocklist/utils.rs

```diff
@@ -85,6 +86,56 @@ pub async fn delete_entry_from_blocklist(
     Ok(blocklist_entry.foreign_into())
 }
 
+pub async fn toggle_blocklist_guard_for_merchant(
+    state: &AppState,
+    merchant_id: String,
+    query: api_blocklist::ToggleBlocklistQuery,
+) -> CustomResult<api_blocklist::ToggleBlocklistResponse, errors::ApiErrorResponse> {
+    let key = get_blocklist_guard_key(merchant_id.as_str());
+    let maybe_guard = state.store.find_config_by_key(&key).await;
+    let new_config = configs::ConfigNew {
+        key: key.clone(),
+        config: query.status.to_string(),
+    };
+    match maybe_guard {
+        Ok(_config) => {
+            let updated_config = configs::ConfigUpdate::Update {
+                config: Some(query.status.to_string()),
+            };
+            state
+                .store
+                .update_config_by_key(&key, updated_config)
+                .await
+                .change_context(errors::ApiErrorResponse::InternalServerError)
+                .attach_printable("Error enabling the blocklist guard")?;
+        }
+        Err(e) if e.current_context().is_db_not_found() => {
+            state
+                .store
+                .insert_config(new_config)
+                .await
+                .change_context(errors::ApiErrorResponse::InternalServerError)
+                .attach_printable("Error enabling the blocklist guard")?;
+        }
+        Err(e) => {
+            logger::error!(error=?e);
+            Err(e)
+                .change_context(errors::ApiErrorResponse::InternalServerError)
+                .attach_printable("Error enabling the blocklist guard")?;
+        }
+    };
+    let guard_status = if query.status { "enabled" } else { "disabled" };
+    Ok(api_blocklist::ToggleBlocklistResponse {
+        blocklist_guard_status: format!("{} {}", "blocklist guard is", guard_status),
+    })
```

Comment:
```suggestion
    Ok(api_blocklist::ToggleBlocklistResponse {
        blocklist_guard_status: guard_status,
    })
```
No need to send back a message. If the merchant wants to make a decision based on the response, it's easier to match on the standalone status as opposed to a message.